File: programmers.co.kr/하샤드 수.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

bool solution(int x) {
    bool answer = true;
    
    int a, b = 0;
    a = x;
    while (a > 0){
        b += (a % 10);
        a /= 10;
    }
    
    if (x % b == 0){
        return answer;
    }else {
        return false;
    }
    return 0;
}

File: programmers.co.kr/하샤드 수.js
function solution(x) {
    var answer = true;
    
    var a = String(x);
    var b = 0;
    for (var i = 0; i < a.length; i++){
        b += Number(a[i]);
    }
    
    if (x % b === 0){
        return answer;
    }else{
        return false;
    }
}